src/core/server/index.ts

```diff
@@ -394,15 +394,12 @@ export {
   mergeSavedObjectMigrationMaps,
 } from '@kbn/core-saved-objects-utils-server';
 export { SavedObjectTypeRegistry } from '@kbn/core-saved-objects-base-server-internal';
-
+export type { SavedObjectsRepository } from '@kbn/core-saved-objects-api-server-internal';
+export { SavedObjectsClient } from '@kbn/core-saved-objects-api-server-internal';
 export type {
-  SavedObjectsRepository,
-  SavedObjectsExporter,
   SavedObjectsExportError,
```

Comment:
`SavedObjectExportError` is public. Is there a reason we don't have public type packages for `import`/`export` ?

Reply:
> SavedObjectExportError is public

Yea, and no... tbh it wasn't supposed to be public, but some consumers started using it, so.. here we are.

`SavedObjectExportError` (and the import counterpart) are concrete implementations atm, without an explicit type equivalent. I basically didn't want to increase the changes of this PR, as introducing a type would mean adapting the existing external usages. 

But I can take another look

>  Is there a reason we don't have public type packages for import/export

I regrouped all the public types of all sub domains in the same `@kbn/core-saved-objects-server` package. Because it was easier, but also because we effectively have some cyclic dependencies between the public types of the subdomains IIRC.

Reply:
ya, the cyclic dependencies would've meant we need somewhere common to share these types from.